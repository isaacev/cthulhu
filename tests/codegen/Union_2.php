<?php

namespace Union_2 {
  abstract class Foo {}
  class Bar extends \Union_2\Foo {
    function __construct($a) {
      $this->{0} = $a;
    }
  }
  function main() {
    new \Union_2\Bar(123);
  }
}

namespace {
  \Union_2\main(null);
}
