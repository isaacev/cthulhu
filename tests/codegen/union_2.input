type Foo =
  | Bar(::Kernel::Int);

#[entry]
fn main() -> () {
  Foo::Bar(123);
}
