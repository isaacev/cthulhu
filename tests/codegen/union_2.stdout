<?php

namespace union_2 {
  abstract class Foo {}
  class Bar extends \union_2\Foo {
    public $fields;
    function __construct($a) {
      $this->fields = [$a];
    }
  }
  // #[entry]
  function main() {
    new \union_2\Bar(123);
  }
}

namespace {
  \union_2\main();
}
